### PR TITLE
Fix for #944

### PR DESCRIPTION
**Pull request by avamsi**
## Related issue number
#944 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes

### CODE REVIEW

**Comment by avamsi** [comment]
@asvetlov Just saw the failing tests. Sorry about that.
If you can take a look at the code changes and confirm everything's okay, I'll update affected tests accordingly.

File: aiohttp/web_reqrep.py [hunk 1]
```diff
@@ -774,7 +774,7 @@ def __init__(self, *, body=None, status=200,
         self.set_tcp_cork(True)
 
         if body is not None and text is not None:
-            raise ValueError("body and text are not allowed together.")
+            raise ValueError('body and text are not allowed together')
```
Comment (asvetlov) [inline aiohttp/web_reqrep.py]: Please use `"` for exception messages

File: aiohttp/web_reqrep.py [hunk 2]
```diff
@@ -794,25 +794,23 @@ def __init__(self, *, body=None, status=200,
                 self._content_dict = {'charset': charset}
                 self.body = text.encode(charset)
             else:
-                self.text = text
                 if content_type or charset:
-                    raise ValueError("Passing both Content-Type header and "
-                                     "content_type or charset params "
-                                     "is forbidden")
+                    raise ValueError('passing both Content-Type header and '
```
Comment (asvetlov) [inline aiohttp/web_reqrep.py]: `"`

File: aiohttp/web_reqrep.py [hunk 3]
```diff
@@ -794,25 +794,23 @@ def __init__(self, *, body=None, status=200,
...
+                self.text = text
         else:
             if hdrs.CONTENT_TYPE in self.headers:
                 if content_type or charset:
-                    raise ValueError("Passing both Content-Type header and "
-                                     "content_type or charset params "
-                                     "is forbidden")
-            if content_type:
-                self.content_type = content_type
+                    raise ValueError('passing both Content-Type header and '
```
Comment (asvetlov) [inline aiohttp/web_reqrep.py]: `"`

File: aiohttp/web_reqrep.py [hunk 4]
```diff
@@ -794,25 +794,23 @@ def __init__(self, *, body=None, status=200,
...
+                                     'content_type or charset params '
+                                     'is forbidden')
+            if content_type is None:
```
Comment (asvetlov) [inline aiohttp/web_reqrep.py]: I think this code should be in `StreamResponse` ctor.
`Content-Type` should be `application/octet-stream` if not explicitly specified.
It's true for all responses.

See https://tools.ietf.org/html/rfc7231#section-3.1.1.5

File: aiohttp/web_reqrep.py (hunk 1 above)
Comment (avamsi) [reply aiohttp/web_reqrep.py]: I thought the difference in quotes usage was due to different contributors. I'll change it back

**Comment by asvetlov** [comment]
Please don't forget about:

> I think this code should be in StreamResponse ctor.
> Content-Type should be application/octet-stream if not explicitly specified.
> It's true for all responses.
> 
> See https://tools.ietf.org/html/rfc7231#section-3.1.1.5

**Comment by asvetlov** [comment]
Thanks!

**Pull request merged**